src/internal/ethers-event.ts

```diff
@@ -0,0 +1,322 @@
+import {
+  AlchemyEventFilter,
+  AlchemyEventType,
+  AlchemyMinedTransactionsAddress,
+  AlchemyMinedTransactionsEventFilter,
+  AlchemyPendingTransactionsEventFilter,
+  AlchemySubscription,
+  NonEmptyArray
+} from '../types/types';
+import {
+  ALCHEMY_EVENT_TYPES,
+  ALCHEMY_MINED_TRANSACTIONS_EVENT_TYPE,
+  ALCHEMY_PENDING_TRANSACTIONS_EVENT_TYPE,
+  Event
+} from './internal-types';
+
+/**
+ * Wrapper class around the ethers `Event` class in order to add support for
+ * Alchemy's custom subscriptions types.
+ *
+ * The serialization and deserialization mechanism requires the order of the
+ * fields to be the same across different {@link ALCHEMY_EVENT_TYPES}. Before
+ * using a getter on a new event filter, make sure that the position of the
+ * field in serialization is correct.
+ *
+ * The getters on this class deserialize the event tag generated by
+ * {@link getAlchemyEventTag} into the original fields passed into the event.
+ */
+export class EthersEvent extends Event {
+  /**
+   * Converts the event tag into the original `fromAddress` field in
+   * {@link AlchemyPendingTransactionsEventFilter}.
+   */
+  get fromAddress(): string | string[] | undefined {
+    const comps = this.tag.split(':');
+    if (comps[0] !== ALCHEMY_PENDING_TRANSACTIONS_EVENT_TYPE) {
+      return undefined;
+    }
+    if (comps[1] && comps[1] !== '*') {
+      return deserializeAddressField(comps[1]);
+    } else {
+      return undefined;
+    }
+  }
+
+  /**
+   * Converts the event tag into the original `toAddress` field in
+   * {@link AlchemyPendingTransactionsEventFilter}.
+   */
+  get toAddress(): string | string[] | undefined {
+    const comps = this.tag.split(':');
+    if (comps[0] !== ALCHEMY_PENDING_TRANSACTIONS_EVENT_TYPE) {
+      return undefined;
+    }
+    if (comps[2] && comps[2] !== '*') {
+      return deserializeAddressField(comps[2]);
+    } else {
+      return undefined;
+    }
+  }
+
+  /**
+   * Converts the event tag into the original `hashesOnly` field in
+   * {@link AlchemyPendingTransactionsEventFilter} and {@link AlchemyMinedTransactionsEventFilter}.
+   */
+  get hashesOnly(): boolean | undefined {
```

Comment:
This code is the tricky part I was referring to in the PR description since it relies on the the `hashesOnly` property being serialized in the 3rd position on both event types.